### PR TITLE
incorrect bits used to determine b's sign - leads to wildly off readings

### PR DESCRIPTION
**Pull request by richardstephens**
While testing on a Dell PowerEdge R630, we noticed temperature readings that were wildly off:
Screenshot 2024-01-16 at 1 24 10 pm

We observed that the value of B used in the reading conversion formula was 896, but the Full sensor record SDR defines B as a 10-bit signed integer. We tracked it down to `FullSensorRecord::parse(...)` using one of the accuracy bits to determine the sign of B.

With this change, we see sensible numbers that are in line with what we get from ipmitool (debug prints added). I'm still investigating why ipmi-rs gives us a bogus value for lower non-recoverable.
Screenshot 2024-01-16 at 1 44 18 pm
Screenshot 2024-01-16 at 1 44 41 pm

### CODE REVIEW

**Comment by richardstephens** [comment]
I wasn't sure the best place to put the tests.

**Comment by datdenkikniet** [comment]
Good catch, thank you! Please fix the comment and I will merge + release :)

> I'm still investigating why ipmi-rs gives us a bogus value for lower non-recoverable.

The bogus value for the lower-non-recoverable value seems to be the lowest value that is supported by the sensor overall. I cannot find where it is defined clearly, but I assume that the minimum value is supposed to signal that the threshold is actually not supported.

File: src/lib.rs [hunk 1]
```diff
@@ -143,3 +143,62 @@ where
         Some(next_record.record)
     }
 }
+
+#[cfg(test)]
```
Comment (datdenkikniet) [inline src/lib.rs:147]: Would prefer if we could put this test in a separate

```
#[cfg(test)]
mod tests;
```

with a separate file

**Review by datdenkikniet — CHANGES_REQUESTED** [review]

**Comment by richardstephens** [comment]
> Would prefer if we could put this test in a...with a separate file

Done, I think we'll probably want to add a few more tests too. I might also look at the broken ones, we're likely going to need to make some changes to RCMP as well.

**Review by datdenkikniet — APPROVED** [review]

**Comment by datdenkikniet** [comment]
Awesome, thanks! Merged, will cook up a release :)

**Pull request merged**